Lint warning

```
./srcs/parser.hpp:165:3: warning: explicitly defaulted default constructor is implicitly deleted [clang-diagnostic-defaulted-function-deleted]
  OpExp() = default;
  ^
./srcs/parser.hpp:154:23: note: default constructor of 'OpExp' is implicitly deleted because base class 'Parser::Exp' has no default constructor
class Parser::OpExp : public Exp {
```

--- a/.github/workflows/c-cpp.yml
+++ b/.github/workflows/c-cpp.yml
@@ -13,7 +13,7 @@ jobs:
       - name: Linter check
         run: |
           pip install cpplint
-          cpplint --filter=-legal/copyright,-runtime/references --linelength=120 --recursive .
+          cpplint --filter=-legal/copyright,-runtime/references --exclude=srcs/libs --linelength=120 --recursive .
       - name: Build
         run: |
           mkdir build

--- a/.github/workflows/c-cpp.yml
+++ b/.github/workflows/c-cpp.yml
@@ -6,6 +6,8 @@ jobs:
     steps:
       - name: Install python dependencies
         uses: actions/setup-python@v1
+      - name: Install clang dependencies
+        run: sudo apt install clang-tidy -y
       - name: Get latest cmake
         uses: lukka/get-cmake@latest
       - name: Checkout source code
@@ -19,6 +21,7 @@ jobs:
           mkdir build
           cd build
           cmake .. -DFRUIT_USES_BOOST=False
+          make tidy
           make
           cd ..
       - name: Test Token

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -79,6 +79,7 @@ if(CLANG_TIDY)
     COMMAND ${CLANG_TIDY}
     ${ALL_CXX_SOURCE_FILES}
     --header-filter=.*
+    --warnings-as-errors=*
     --
     --std=c++17
     ${INCLUDE_HEADER_DIR}

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,18 +18,18 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_SOURCE_DIR}/bin)
 set(CMAKE_BUILD_TYPE Release)
 set(FRUIT_TESTS_USE_PRECOMPILED_HEADERS OFF)
 set(SRCS
-srcs/calcdfafactory.cc
-srcs/dfa.cc
-srcs/token.cc
-srcs/tokenimpl.cc
-srcs/visitor.cc
-srcs/dfabuilder.cc
-srcs/parser.cc
-srcs/tokengen.cc
-srcs/validator.cc
+${PROJECT_SOURCE_DIR}/srcs/calcdfafactory.cc
+${PROJECT_SOURCE_DIR}/srcs/dfa.cc
+${PROJECT_SOURCE_DIR}/srcs/token.cc
+${PROJECT_SOURCE_DIR}/srcs/tokenimpl.cc
+${PROJECT_SOURCE_DIR}/srcs/visitor.cc
+${PROJECT_SOURCE_DIR}/srcs/dfabuilder.cc
+${PROJECT_SOURCE_DIR}/srcs/parser.cc
+${PROJECT_SOURCE_DIR}/srcs/tokengen.cc
+${PROJECT_SOURCE_DIR}/srcs/validator.cc
 )
-set (LIB_HEADER_DIR "") # placeholder for now
-
+set(ALL_CXX_SOURCE_FILES ${SRCS} ${PROJECT_SOURCE_DIR}/srcs/main.cc)
+set(LIB_HEADER_DIR "") # placeholder for now
 # Find packages go here
 find_package(Git QUIET)
 if(GIT_FOUND AND EXISTS "${PROJECT_SOURCE_DIR}/.git")
@@ -53,14 +53,41 @@ endif()
 
 
 
+
+
 add_subdirectory("${PROJECT_SOURCE_DIR}/libs/fruit")
 
 
 set(FRUIT_HEADER_DIR "${PROJECT_SOURCE_DIR}/libs/fruit/include/" "${Fruit_BINARY_DIR}/include/")
+set(INCLUDE_HEADER_DIR)
 
-add_executable(calc srcs/main.cc ${SRCS})
-add_executable(testToken tests/testToken.cc ${SRCS})
-add_executable(testParser tests/testParser.cc ${SRCS})
+# Only Available in CMake3.12+
+if (${CMAKE_VERSION} VERSION_GREATER_EQUAL 3.12)
+    set(INCLUDE_HEADER_DIR ${FRUIT_HEADER_DIR})
+    list(TRANSFORM INCLUDE_HEADER_DIR PREPEND -I)
+else()
+    foreach(DIR in FRUIT_HEADER_DIR)
+        list(APPEND INCLUDE_HEADER_DIR -I${DIR})
+    endforeach(DIR in FRUIT_HEADER_DIR)
+endif()
+
+# Adding clang-tidy target if executable is found
+find_program(CLANG_TIDY "clang-tidy")
+if(CLANG_TIDY)
+  add_custom_target(
+    tidy
+    COMMAND ${CLANG_TIDY}
+    ${ALL_CXX_SOURCE_FILES}
+    --header-filter=.*
+    --
+    --std=c++17
+    ${INCLUDE_HEADER_DIR}
+    )
+endif()
+
+add_executable(calc ${PROJECT_SOURCE_DIR}/srcs/main.cc ${SRCS})
+add_executable(testToken ${PROJECT_SOURCE_DIR}/tests/testToken.cc ${SRCS})
+add_executable(testParser ${PROJECT_SOURCE_DIR}/tests/testParser.cc ${SRCS})
 set_target_properties(calc testToken testParser PROPERTIES CXX_STANDARD 17)
 target_include_directories(calc PRIVATE "${FRUIT_HEADER_DIR}" PUBLIC "${LIB_HEADER_DIR}")
 target_include_directories(testToken PRIVATE "${FRUIT_HEADER_DIR}" PUBLIC "${LIB_HEADER_DIR}")

--- a/srcs/parser.hpp
+++ b/srcs/parser.hpp
@@ -162,7 +162,6 @@ class Parser::OpExp : public Exp {
  protected:
   OpExp(std::size_t pos, std::initializer_list<std::shared_ptr<Exp>> list)
       : Exp{ExpType::OP, pos}, _children{list} {}
-  OpExp() = default;
 
  private:
   using ExpIt = std::vector<std::shared_ptr<Exp>>::iterator;

